### PR TITLE
Updating Azure.Identity xml doc comments

### PR DESCRIPTION
**Pull request by schaabs**

### CODE REVIEW

**Review by AlexGhiondea — APPROVED** [review]

File: sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs [hunk 1]
```diff
@@ -56,7 +56,8 @@ internal DefaultAzureCredential(DefaultAzureCredentialFactory factory, DefaultAz
         }
 
         /// <summary>
-        /// Sequencially calls <see cref="TokenCredential.GetToken"/> on all the specified sources, returning the first successfully retured <see cref="AccessToken"/>.
+        /// Sequentially calls <see cref="TokenCredential.GetToken"/> on all the included credentials in the order <see cref="EnvironmentCredential"/>, <see cref="ManagedIdentityCredential"/>, <see cref="SharedTokenCacheCredential"/>,
+        /// and <see cref="InteractiveBrowserCredential"/> returning the first successfully obtained <see cref="AccessToken"/>. This method is called by Azure SDK clients. It isn't intended for use in application code.
```
Comment (ellismg) [inline sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs]: FWIW - `InteractiveBrowserCredential` is not called by default (unless `includeInteractiveCredentials` is set to `true` when constructing the `DefaultAzureCredential`, unsure if we want to call that out or not.

File: sdk/identity/Azure.Identity/src/EnvironmentCredential.cs [hunk 2]
```diff
@@ -10,15 +10,17 @@
...
-    /// Enables authentication to Azure Active Directory using client secret
+    /// Enables authentication to Azure Active Directory using client secret, or username and password,
     /// details configured in the following environment variables:
     /// <list type="table">
     /// <listheader><term>Variable</term><description>Description</description></listheader>
     /// <item><term>AZURE_TENANT_ID</term><description>The Azure Active Directory tenant(directory) ID.</description></item>
     /// <item><term>AZURE_CLIENT_ID</term><description>The client(application) ID of an App Registration in the tenant.</description></item>
     /// <item><term>AZURE_CLIENT_SECRET</term><description>A client secret that was generated for the App Registration.</description></item>
+    /// <item><term>AZURE_USERNAME</term><description>The username, also known as upn, of an Azure Active Directory user account.</description></item>
+    /// <item><term>AZURE_USERNAME</term><description>The password of the Azure Active Directory user account. Note this does not support accounts with MFA enabled.</description></item>
```
Comment (ellismg) [inline sdk/identity/Azure.Identity/src/EnvironmentCredential.cs]: Should be `AZURE_PASSWORD`?

File: sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs [hunk 3]
```diff
@@ -67,7 +68,8 @@ public override AccessToken GetToken(TokenRequestContext requestContext, Cancell
...
```
Comment (ellismg) [inline sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs]: Similar feedback about `InteractiveBrowserCredetial` here.

**Review by ellismg — CHANGES_REQUESTED** [review]
Some small stuff that jumped out at me.

**Pull request merged**